src/Orleans/Core/ClusterClient.cs

```diff
@@ -143,31 +164,36 @@ public static ClusterClient Create(FileInfo configFile)
         public static ClusterClient Create(ClientConfiguration configuration) => new ClusterClient(configuration);
 
         /// <inheritdoc />
-        public void Start()
+        public async Task Start()
         {
-            lock (this.initLock)
+            this.ThrowIfDisposed();
+            using (await this.initLock.LockAsync())
```

Comment:
Not really, having it in the parent means it will not capture it at that point to continue in the context once the nested continuations finish.
But even without nesting, and just having a flat execution path, adding .ConfigureAwait(false) on the first call is not good enough, because it might continue synchronously (inline), so it didn't capture the context *explicitly*, but it is *still* running in that context.